src/backend/api/meals.js

```diff
@@ -2,14 +2,97 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+//-------------------------------------------
+
+//http://localhost:5000/api/meals/title
+router.get("/title", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
+    const titles = await knex("meal").select("title");
     response.json(titles);
   } catch (error) {
     throw error;
   }
 });
 
+//-------------------------------------------
+
+// GET /api/meals  - Returns all meals
+router.get("/", async (req, res) => {
+  try {
+    const allMeals = await knex.select("*").from("meal");
+    if (allMeals) {
+      res.status(200).json(allMeals);
+    } else {
+      res.status(404).send("The meals were not found");
+    }
+  } catch (error) {
+    console.error("Server Error", err);
+    res.status(500).json({ error: "Server Error: Not able to fetch data" });
+  }
+});
+
+// POST /api/meals - Adds a new meal to the database
+router.post("/", async (req, res) => {
+  try {
+    const newMeal = req.body;
+    const newMealAdded = await knex("meal").insert(newMeal);
+    if (newMealAdded) {
+      res.status(201).json({ message: "The meal was created successfully" });
+    }
+    res.status(404).send("The data you have requested is not found");
```

Comment:
Here I am not sure if the response should be 404, since this is a post request, so we create a new object, so what may go wrong when we try to add something new to the DB?